crates/redis_interface/src/commands.rs

```diff
@@ -148,12 +148,17 @@ impl super::RedisConnectionPool {
     }
 
     #[instrument(level = "DEBUG", skip(self))]
-    pub async fn delete_key(&self, key: &str) -> CustomResult<(), errors::RedisError> {
-        self.pool
-            .del(key)
-            .await
-            .into_report()
-            .change_context(errors::RedisError::DeleteFailed)
+    pub async fn delete_key(&self, key: &str) -> CustomResult<DelReply, errors::RedisError> {
+        let result: Result<DelReply, fred::error::RedisError> = self.pool.del(key).await;
+        match result {
+            Ok(del_reply) => Ok(del_reply),
+            Err(e) => match e.kind() {
+                fred::error::RedisErrorKind::NotFound => {
+                    Err(errors::RedisError::NotFound).into_report()
+                }
+                _ => Err(errors::RedisError::DeleteFailed).into_report(),
+            },
+        }
```

Comment:
```suggestion
        self.pool.del(key).await.into_report().change_context(errors::RedisError::DeleteFailed)
```

You should be able to do something similar, since the `NotFound` error is never raised in the current implementation.